### ISSUE TITLE
Support filtering builds by creation datetime in `/api/v3/projects/(str: project_slug)/builds/` API v3 endpoint

### ISSUE THREAD

**Issue by edgarrmondragon** [readthedocs/readthedocs.org#8971]
## Details

* Read the Docs project URL: https://readthedocs.org/projects/citric/
* Build URL (if applicable): N/A
* Read the Docs username (if applicable):  edgarrm92

## Expected Result

Adding a query parameter (e.g. `created_after`) would allow to get only the builds created after a certain date.

## Actual Result

All builds (paginated) are returned.

**Comment by edgarrmondragon** [comment]
Hi @humitos! Sure. I'm trying to use the data exposed by the API to perform some basic analytics in an ELT framework, which essentially means extracting raw data from the source system (RTD) and loading directly in a data warehouse.

One common performance improvement for endpoints in front of large datasets is to extract the data _incrementally_, meaning one stores a bookmark (e.g. the largest seen value of `updated_at`) so in the next run of the extraction only the most recent records are retrieved. Some endpoints, like `/api/v3/projects/`, return a small number of records in most cases so it's ok to get them all every time, but others like `/api/v3/projects/{project_slug}/builds` usually return more records, depending on the level of activity in the project, and it makes sense to filter them by recency.

Wdyt? Do you think this is a common or interesting enough use case that it warrants a solution? I'd also be willing to work on a PR of that helps 🙂

**Comment by humitos** [comment]
Thanks for sharing your use case. We will discuss this internally and make a decision. However, I don't think this will happen _soon_ because our roadmap is overloaded with a lot of tasks currently.

However, since that endpoint is sorted by `created` date and paginated, you can probably implement exactly what you need without requiring a filter in the server: store the last `created` date on your side and paginate the results until you find that one. I think that could work in the meanwhile. What do you think?

**Comment by edgarrmondragon** [comment]
Yeah, the endpoint is sorted by `created` descending, so that's a feasible workaround and I think I will go with it for now.

Thanks for the feedback, @humitos!
